### PR TITLE
[next] Watcher 2.0

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- More reliable process killing under `linux`, `mac` and `windows`. Inspired by `nodemon` kill functionality.
- Implement `sane` watcher backend, which is the same as the `jest` watcher backend.
- Automatically detect `watchman` and turn on watchman support in `sane`.
- Use `sane` watcher as default watcher.
- Improved and fixed watcher and executor cleanup and initialisation
- Support watcher `sets` and `defaultSet`
- Changed `docz` default port to `3030` so it doesn't conflict with the server.
- Upgrade deps
- `watcher` bin signature is now: `watcher --config <configFile> [sets or watches] ...`

## How do I test this PR?

- `npm run watch`
- `npm run watch -- server`
- `npm run watch -- client`
- `npm run watch -- docz`

### CODE REVIEW

File: config/webpackDevServer.config.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const paths = require("./paths");
 const protocol = process.env.HTTPS === "true" ? "https" : "http";
 const host = process.env.HOST || "0.0.0.0";
 const serverPort = process.env.PORT || 3000;
```
Comment (wyattjoh) [inline config/webpackDevServer.config.js:11]: At some point, we need to harmonize what the client uses to read config with what the server is using.

File: scripts/watcher/LongRunningExecutor.ts [hunk 2]
```diff
@@ -51,18 +50,43 @@ export default class LongRunningExecutor implements Executor {
     });
   }
 
-  private restart(): void {
+  private async restart(): Promise<void> {
```
Comment (wyattjoh) [inline scripts/watcher/LongRunningExecutor.ts]: Why define the return type here while it's can be calculated?

File: scripts/watcher/pstree.remy.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+declare module "pstree.remy" {
```
Comment (wyattjoh) [inline scripts/watcher/pstree.remy.d.ts]: Why not move this into a `types` folder under `scripts/watcher`?

File: config/webpackDevServer.config.js (hunk 1 above)
Comment (alinhle) [reply config/webpackDevServer.config.js:11]: Yes we should, step by step :-)

File: scripts/watcher/pstree.remy.d.ts (hunk 3 above)
Comment (alinhle) [reply scripts/watcher/pstree.remy.d.ts]: Sure

File: scripts/watcher/LongRunningExecutor.ts (hunk 2 above)
Comment (alinhle) [reply scripts/watcher/LongRunningExecutor.ts]: Sure

**Review by wyattjoh — APPROVED** [review]
Looks good to me!

**Review by wyattjoh — DISMISSED** [review]
seems to not stop restarting??

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**